### PR TITLE
Factor new GetClusterNodes() out of CreateVolume().

### PR DESCRIPTION
**Pull request by humblec**
Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

### CODE REVIEW

**Comment by humblec** [comment]
@jsafrane @rootfs PTAL.

**Comment by gnufied** [comment]
@k8s-bot test this

**Comment by humblec** [comment]
@k8s-bot test this

**Comment by jsafrane** [comment]
@k8s-bot test this

**Comment by humblec** [comment]
@jsafrane Thanks :)..

**Comment by humblec** [comment]
@thockin can you please take a look at this PR ?

**Comment by jsafrane** [comment]
@k8s-bot test this

**Comment by humblec** [comment]
Looks to be a spurious failure.. I will resend the patchset and trigger the tests again. Thanks @jsafrane

**Comment by jsafrane** [comment]
@k8s-bot test this

File: pkg/volume/glusterfs/glusterfs.go [hunk 1]
```diff
@@ -96,7 +96,6 @@ func (plugin *glusterfsPlugin) GetVolumeName(spec *volume.Spec) (string, error)
 	if err != nil {
 		return "", err
 	}
-
```
Comment (obnoxxx) [inline pkg/volume/glusterfs/glusterfs.go]: First of all, these empty-line removals are not related to the change described in the commit message, and should not be sneaked in. *If* they are to be applied, they should be a patch of their own.

Secondly, I put several of those newlines into the text to increase readability. I think grouping blocks and lines of code by inserting newlines *greatly* increases the readability of code. I therefore request most of them to remain. If there is a "formatting policy I don't know of", I would be willing to disuss... :-)

File: pkg/volume/glusterfs/glusterfs.go [hunk 2]
```diff
@@ -143,6 +141,7 @@ func (plugin *glusterfsPlugin) NewMounter(spec *volume.Spec, pod *v1.Pod, _ volu
 }
 
 func (plugin *glusterfsPlugin) getGlusterVolumeSource(spec *volume.Spec) (*v1.GlusterfsVolumeSource, bool) {
+
```
Comment (obnoxxx) [inline pkg/volume/glusterfs/glusterfs.go]: This newline after the function head is imho extra and should *not* be added.

File: pkg/volume/glusterfs/glusterfs.go [hunk 3]
```diff
@@ -765,7 +739,6 @@ func (p *glusterfsVolumeProvisioner) CreateVolume(gid int) (r *v1.GlusterfsVolum
 }
 
 func (p *glusterfsVolumeProvisioner) createEndpointService(namespace string, epServiceName string, hostips []string, pvcname string) (endpoint *v1.Endpoints, service *v1.Service, err error) {
-
```
Comment (obnoxxx) [inline pkg/volume/glusterfs/glusterfs.go]: ack. on this removal. but in a different commit.

File: pkg/volume/glusterfs/glusterfs.go [hunk 4]
```diff
@@ -680,16 +648,40 @@ func (r *glusterfsVolumeProvisioner) Provision() (*v1.PersistentVolume, error) {
...
+		ipaddr := dstrings.Join(nodei.NodeAddRequest.Hostnames.Storage, "")
+		dynamicHostIps = append(dynamicHostIps, ipaddr)
+	}
+	glog.V(3).Infof("glusterfs: hostlist :%v", dynamicHostIps)
+	if len(dynamicHostIps) == 0 {
+		glog.Errorf("glusterfs: no hosts found: %v", err)
+		return nil, fmt.Errorf("no hosts found: %v", err)
+	}
+	return dynamicHostIps, nil
+}
```
Comment (obnoxxx) [inline pkg/volume/glusterfs/glusterfs.go]: empty line after the funtion is missing.

**Review by obnoxxx — CHANGES_REQUESTED** [review]
The actual content looks good. But I suggest the following changes to be made:
- I suggest a slightly different commit message: IMHO It should read something like "Factor new GetClusterNodes() out of CreateVolume()".
- The newline removals (and additions) are not related to the description of this patch and should be a different patch (if at all). We should not sneak in changes under the hood!
- I would argue that the newlines were added for readability and should remain, except for very few that I marked in my comment.

**Comment by jsafrane** [comment]
@k8s-bot non-cri e2e test this
@k8s-bot gci gce e2e test this

File: pkg/volume/glusterfs/glusterfs.go (hunk 4 above)
Comment (humblec) [inline pkg/volume/glusterfs/glusterfs.go]: Done.

**Comment by humblec** [comment]
Addressed the comments. I will send a separate PR for empty line cleanup.

**Comment by obnoxxx** [comment]
Thanks @humblec , this patch looks clean now to me.

**Comment by jsafrane** [comment]
@k8s-bot test this

**Comment by humblec** [comment]
@jsafrane the unit integration failure looks be a spurious one to me. :(

**Comment by jsafrane** [comment]
@k8s-bot unit test this

**Comment by humblec** [comment]
@jsafrane Thanks.. All the tests are passed.!

**Pull request merged**